Src/DependencyCollector/Shared/Implementation/HttpDesktopDiagnosticSourceSubscriber.cs

```diff
@@ -48,6 +48,7 @@ public void OnNext(DiagnosticListener value)
                 if (value.Name == "System.Net.Http.Desktop")
                 {
                     this.sourceSubscription = value.Subscribe(this.parent, (Predicate<string>)null);
+                    DependencyTableStore.Instance.IsDesktopHttpDiagnosticSourceActivated = true;
```

Comment:
Added checks in DependencyTrackingTelemetryModuleTestNet46.

There is no guarantee that subscription will happen before module initialization. It must happen before the first HTTP outgoing call or will never happen at all.
So if there is a HTTP call, in EventSource we will know for sure if diag source is activated or not